pkg/exec/err_max_attempts.go

```diff
@@ -0,0 +1,36 @@
+package exec
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+type ErrAttemptsExceeded struct {
```

Comment:
Do we really need separate files for separate errors?
Usually, you would have a lot larger files in the stdlib

Reply:
I don't like large files :D